ProjectSourceCode/src/index.js

```diff
@@ -454,6 +454,103 @@ app.get('/viewReviews', async (req, res) => {
 }
  });
 
+
+ app.get('/ownReviews',auth,async (req,res) => {
+  if(!req.session||!req.session.user||!req.session.user.username){
+    console.log('no user session found');
+    return res.redirect('/login');
+  }
+
+    const username= req.session.user.username;
+    console.log('Session: ',req.session);
+    console.log('username: ',username);
+    try{
+      console.log('STEP1')
+      const reviews = await db.any(
+      `SELECT r.review_id, r.review, r.rating, s.seat_number, s.section, s.row, e.event_name, e.event_date, u.username
+      FROM reviews r
+      JOIN reviews_to_events re ON r.review_id = re.review_id
+      JOIN events e ON re.event_id = e.event_id
+      JOIN reviews_to_seats rs ON r.review_id = rs.review_id
+      JOIN seats s ON rs.seat_id = s.seat_id
+      JOIN reviews_to_users ru ON r.review_id = ru.review_id
+      JOIN users u ON u.username =ru.username
+      WHERE u.username = $1`, [username]);
+      console.log('reviews: ', reviews)
+    if(reviews.length==0){
+      console.log('STEP3')
+      res.render('pages/viewReviews', {
+        message: "You don't have any reviews yet!"
+      });
+      console.log('STEP4')
+    }
+    else{
+      console.log('STEP5')
+      res.render('pages/OwnReviews', { reviews: reviews });
+    }
+  }
+  catch(err){
+    console.log('STEP6')
+    console.error('error finding ownReviews: ',err.message)
+  }
+});
+
+app.post('/deleteReview/:reviewID',auth,async (req,res) => {
+    const reviewID = req.params.reviewID;
+    try{
+      await db.none('DELETE FROM reviews_to_events WHERE review_id=$1',[reviewID]);
+      await db.none('DELETE FROM reviews_to_seats WHERE review_id=$1',[reviewID]);
+      await db.none('DELETE FROM reviews_to_users WHERE review_id=$1',[reviewID]);
+
+      await db.none('DELETE FROM reviews WHERE review_id=$1',[reviewID]);
+
+      res.redirect('/ownReviews')
+    }
+    catch(err){
+      console.error('error deleting review:',err.message)
+    }
+});
+
+app.get('/editReview/:reviewID',auth,async (req,res) => {
+  const reviewID=req.params.reviewID;
+  try{
+    const review= await db.one('SELECT * FROM reviews WHERE review_id=$1',[reviewID]);
+
+    const seat = await db.one('SELECT * FROM seats s ' + 'JOIN reviews_to_seats rs ON s.seat_id = rs.seat_id ' + 'WHERE rs.review_id=$1',[reviewID]);
+
+    const event = await db.one('SELECT * FROM events e ' + 'JOIN reviews_to_events re ON e.event_id = re.event_id ' + 'WHERE re.review_id=$1',[reviewID]);
+
+    res.render('pages/editReview',{
+      review: review,
+      reviewID: reviewID,
+      seat: seat,
+      event: event
+    });
+  }
+  catch(err){
+    console.error('error in get editReview')
+  }
+});
+
+app.post('/editReview/:reviewID',auth,async (req,res) => {
+  const reviewID=req.params.reviewID;
```

Comment:
this would also let any user send a post request to edit any review as long as their logged in to somebody